src/java/main/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -399,18 +403,20 @@ public boolean truncate(long zxid) throws IOException {
             }
             long pos = input.getPosition();
             // now, truncate at the current position
-            RandomAccessFile raf=new RandomAccessFile(itr.logFile,"rw");
+            raf = new RandomAccessFile(itr.logFile, "rw");
             raf.setLength(pos);
-            raf.close();
             while(itr.goToNextLog()) {
-                if (!itr.logFile.delete()) {
-                    LOG.warn("Unable to truncate {}", itr.logFile);
+                try {
+                    Files.delete(itr.logFile.toPath());
+                } catch (NoSuchFileException e) {
                 }
             }
```

Comment:
- Do we have any intentions for changing `itr.logFile.delete() `  to ` Files.delete(itr.logFile.toPath())` ? confused about this:
   > Files.delete can throw an exception, which this is useful for error reporting and to diagnosed why a file cannot be deleted.
- catch `NoSuchFileException` with even no log ?

Reply:
- i think ```Files.delete```  can provide more detailed messages if delete operation failed.  I'm not sure this change is necessary or not.
- update

Reply:
- A option:  remain the origin code in the master branch unchanged.just log the `itr.logFile` and `zxid`
- B option:  choose `Files.delete`  because it throws `exception` rather than `itr.logFile.delete()` returns `boolean` ,**dont't catch it** and it will be caught in `Learner.java#syncWithLeader` and log it with `zxid`
- am I right?

Reply:
update